source/AI.cpp

```diff
@@ -2654,6 +2663,41 @@ void AI::DoScatter(Ship &ship, Command &command)
 
 
 
+bool AI::DoSecretive(Ship &ship, Command &command)
+{
+	shared_ptr<Ship> scanningShip;
+	// Figure out if any ship is currently scanning us. If that is the case, move away from it.
+	for(auto otherShip : GetShipsList(ship, false))
+		if(!ship.GetGovernment()->Trusts(otherShip->GetGovernment()) &&
+				otherShip->Commands().Has(Command::SCAN) &&
+				otherShip->GetTargetShip() == ship.shared_from_this() &&
+				!otherShip->IsDisabled() && !otherShip->IsDestroyed())
+			scanningShip = make_shared<Ship>(*otherShip);
+
+	if(scanningShip)
+	{
+		Point scanningPos = scanningShip->Position();
+		Point pos = ship.Position();
+
+		double cargoDistance = 100. * sqrt(scanningShip->Attributes().Get("cargo scan power"));
+		double outfitDistance = 100. * sqrt(scanningShip->Attributes().Get("outfit scan power"));
+
+		double maxScanRange = cargoDistance > outfitDistance ? cargoDistance : outfitDistance;
+		double distance =scanningPos.Distance(pos);
```

Comment:
```suggestion
		double distance = scanningPos.Distance(pos);
```
Actually, skip the square roots in calculating cargo and outfit scan ranges and use `DistanceSquared` or whatever the method is called here.

Reply:
Oh yeah good idea, I wasn't aware you could do that before I asked in the hub and got an affirmative answer.